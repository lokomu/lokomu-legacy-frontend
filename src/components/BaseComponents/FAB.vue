<template>
  <div
    class="flex cursor-pointer bg-green-500 hover:bg-green-600 active:bg-green-700 dark:bg-gradient-to-t dark:from-green-500 dark:to-floyd-600 dark:hover:bg-gradient-to-tr dark:shadow-green-500/40 dark:shadow-lg rounded-md rotate-45"
    tabindex="0"
  >
    <PlusIcon class="h-8 fill-white m-auto rotate-45" />
  </div>
</template>

<script setup lang="ts">
import PlusIcon from "@/components/IconComponents/PlusIcon.vue";
</script>
