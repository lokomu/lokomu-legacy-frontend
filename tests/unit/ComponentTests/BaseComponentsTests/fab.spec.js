import { shallowMount } from "@vue/test-utils";
import FAB from "@/components/BaseComponents/FAB";
import { describe, expect, it} from "vitest";

describe("FAB component", () => {
  let wrapper = shallowMount(FAB);

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
