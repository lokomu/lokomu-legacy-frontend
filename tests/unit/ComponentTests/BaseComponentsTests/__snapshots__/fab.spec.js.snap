// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FAB component > renders correctly 1`] = `
<div
  class="flex cursor-pointer bg-green-500 hover:bg-green-600 active:bg-green-700 dark:bg-gradient-to-t dark:from-green-500 dark:to-floyd-600 dark:hover:bg-gradient-to-tr dark:shadow-green-500/40 dark:shadow-lg rounded-md rotate-45"
  tabindex="0"
>
  <plus-icon-stub
    class="h-8 fill-white m-auto rotate-45"
  />
</div>
`;
